src/main/java/com/google/cloud/tools/appengine/AppEngineDescriptor.java

```diff
@@ -99,24 +106,90 @@ public boolean isJava8() {
             && getRuntime().startsWith("java8");
   }
 
-  private static String getTopLevelValue(Document doc, String parentTagName, String childTagName) {
-    try {
-      NodeList parentElements = doc.getElementsByTagNameNS(APP_ENGINE_NAMESPACE, parentTagName);
-      if (parentElements.getLength() > 0) {
-        Node parent = parentElements.item(0);
-        if (parent.hasChildNodes()) {
-          for (int i = 0; i < parent.getChildNodes().getLength(); ++i) {
-            Node child = parent.getChildNodes().item(i);
-            if (child.getNodeName().equals(childTagName)) {
-              return child.getTextContent();
+  /**
+   * Given the following structure:
+   * <pre>
+   * {@code
+   * <env-variables>
+   *   <env-var name="key" value="val" />
+   * </env-variables>
+   * }
+   * </pre>
+   * This will construct a map of the form {[key, val], ...}
+   *
+   * @return a map representing the environment variable settings in the appengine-web.xml.
+   */
+  public Map<String, String> getEnvironment() {
+    Node environmentParentNode = getNode(document, "appengine-web-app", "env-variables");
+    return getAttributeMap(environmentParentNode, "env-var", "name", "value");
+  }
+
+  private static String getText(Node node) {
+    if (node != null) {
+      try {
+        return node.getTextContent();
+      } catch (DOMException ex) {
+        throw new AppEngineException("Failed to parse text content from node "
+            + node.getNodeName());
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Returns a map formed from the attributes of the nodes contained within the parent node.
```

Comment:
then the last one found will be added to the map (according to the contract of Map#put)